examples/clusterclasses/docker/rke2/clusterclass-docker-rke2.yaml

```diff
@@ -71,6 +77,24 @@ spec:
               path: /spec/template/spec/customImage
               valueFrom: 
                 variable: dockerImage
+    - name: dockerAuthSecret
+      definitions:
+        - selector:
+            apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+            kind: RKE2ControlPlaneTemplate
```

Comment:
You need to make similar changes to RKE2ConfigTemplate as well. [See](https://github.com/rancher/turtles/pull/1432/files#diff-96e26eb7d3c2785c3242e74b73485089bc86823ba016350b2cf6e23992a81c56R332).